src/components/DuosDatePicker.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import {createTheme, ThemeProvider} from '@mui/material/styles';
+import {
+  DesktopDatePicker,
+  LocalizationProvider,
+  PickersActionBarProps,
+  PickersDay,
+  PickersDayProps
+} from '@mui/x-date-pickers';
+import {AdapterDayjs} from '@mui/x-date-pickers/AdapterDayjs';
+import {Button} from '@mui/material';
+import DialogActions from '@mui/material/DialogActions';
+import {Dayjs} from 'dayjs';
+import type {} from '@mui/x-date-pickers/themeAugmentation';
+
+
+interface DUOSDatePickerProps {
+    inputFormat: string;
+    value: Dayjs;
+    onChange: Function;
+    onError: Function;
+    readOnly: boolean;
+}
+
+export const DuosDatePicker = (props: DUOSDatePickerProps) => {
+  const {inputFormat, value, onChange, onError, readOnly} = props;
+  const theme = createTheme({
+    palette: {
+      primary:{
+        main:'#216FB4',
+      },
+      secondary:{
+        main:'#ffffff',
+        contrastText:'#216FB4'
+      },
+    },
+    components: {
+      MuiButton:{
+        styleOverrides:{
+          text:{
+            fontFamily: 'Montserrat',
+            fontSize: '13px',
+            fontWeight: '400',
+            padding: '7px 20px 7px 20px',
+          },
+        },
+      },
+      MuiIconButton: {
+        styleOverrides: {
+          sizeMedium: {
+            color:'#216FB4',
+          },
+        },
+      },
+      MuiDayCalendar:{
+        styleOverrides:{
+          weekDayLabel:{
+            fontFamily: 'Montserrat',
+            color: '#000',
+            fontSize: '13px',
+            fontWeight: '400',
+            padding: '7px 20px 7px 20px',
+            '&:first-of-type': {
+              color: 'red',
+            },
+            '&:last-of-type': {
+              color: 'red',
+            },
+          },
+        },
+      },
+      MuiPickersArrowSwitcher:{
+        styleOverrides:{
+          button:{
+            fontSize:'2.5rem',
+          },
+        },
+      },
+      MuiPickersDay: {
+        styleOverrides: {
+          root: {
+            '--weekend': 'red',
+            borderRadius: '8px',
+            fontFamily: 'Montserrat',
+            color: '#000',
+            fontSize: '13px',
+            fontWeight: '400',
+            padding: '7px 20px 7px 20px',
+            '&.Mui-selected': {
+              backgroundColor: '#216FB4',
+              '&:hover': {
+                backgroundColor: '#216FB4',
+              },
+              '&:focus': {
+                backgroundColor: '#216FB4',
+              },
+            },
+          },
+        },
+      },
+      MuiPickersYear: {
+        styleOverrides: {
+          yearButton: {
+            fontFamily: 'Montserrat',
+            color: '#000',
+            fontSize: '13px',
+            fontWeight: '400',
+            padding: '7px 20px 7px 20px',
+            '&.Mui-selected': {
+              backgroundColor: '#216FB4',
```

Comment:
Could we make an intermediate variable for this?

Reply:
@s-rubenstein - Do you mean a CSS variable that calculates the value or a variable that assigns the backgroundColor to the theme?